README.md

```diff
@@ -1,5 +1,88 @@
 Discovery
 =========
 
-TODO
+Weave Discovery retrieves and watches a list of peers
+available in a Weave network from a persistent storage.
+Discovery uses this information for telling a Weave Router
+about new peers it must connect to as well as peers
+that are not available anymore and must be forgotten.
+Discovery will also advertise the Weave router in the
+same backend so other peers can join it.
+
+The storage backends used by Discovery are compatible
+with [Docker Swarm](https://github.com/docker/swarm), so
+you could use Discovery for keeping a Swarm connected
+with a Weave network.
+
+
+Installation
+============
+
+Just download the `discovery` script from the repository,
+or use this short URL:
+
+```
+curl -O http://git.io/vmW3z
+```
+
+This will download the `discovery` script to the current
+directory.
+
+Usage
+=====
+
+Once you have the script, joining a new discovery URL
+is as simple as running:
+
+```
+$ ./discovery join <URL>
+```
+
+The endpoint URL can be something like
+
+- `etcd://<etcd_ip>/<path>`
+- `consul://<consul_addr>/<path>`
+- `zk://<zookeeper_addr1>,<zookeeper_addr2>/<path>`
+- `token://<swarm_cluster_id>`
+
+Example:
+
+```
+$ ./discovery join etcd://192.168.9.1/mycluster
+```
+
+You can get more information about the supported backends
+as well as the URL formats [here](https://github.com/docker/swarm/tree/master/discovery).
+
+By default, Discovery will use the Weave router in the
+local machine. However, you can point Discovery to a
+different router with the `--weave` flag.
+
+For example, we can point Discovery to the Weave HTTP
+API found at _192.168.9.3:6784_ with:
+
+```
+$ ./discovery join --weave=192.168.9.3:6784 etcd://192.168.9.1/mycluster
+```
+
+The router will also be advertised to other peers by
+Discovery by publishing a reachable IP in the same discovery
+endpoint, so other peers will be able to find it.
+Discovery guess the advertised IP by trying _1)_ the
```

Comment:
"guesses"

and what's with the underscores (highlight) around _1)_? You don't have them around 2)?

What does "trying" mean? Trying to connect?